File: pipeline/templates/save_data.r
suppressMessages(library(readr))
library(tools)


save_data <- function(df, path){
    if (file_ext(path) == "feather") {
        write_feather(df, path)
    } else if (file_ext(path) %in% c("csv", "")) {
        write_csv(df, path)
    } else if (file_ext(path) == "rds") {
        saveRDS(df, path)
    } else {
        stop("NotImplementedError")
    }

    return(df)
}

File: pipeline/templates/save_data.py
from pathlib import Path


def save_data(df, path):
    path = Path(path)

    if path.suffix == ".feather":
        df.to_feather(path)
    elif path.suffix == ".dta":
        df.to_stata(path)
    elif path.suffix == ".csv" or path.suffix == "":
        df.to_csv(path)
    elif path.suffix in [".pkl", ".pickle"]:
        df.to_pickle(path)
    else:
        raise NotImplementedError

    return df
